From 745c69bc3120dff0cede965ed633bf0d74611935 Mon Sep 17 00:00:00 2001
From: Waiman Long <Waiman.Long@hp.com>
Date: Mon, 12 May 2014 16:58:58 -0400
Subject: [PATCH 2/5] mutex: Enable the queuing of mutex spinners with MCS lock
Patch-mainline: Never, SLES 11 SP3-specific
References: FATE#317271

This patch enables the use of MCS-style lock for queuing the mutex
spinnner. As this patch can cause kABI breakage, it can be disabled, if
necessary, by taking it out without affecting other part of the code.

Signed-off-by: Waiman Long <Waiman.Long@hp.com>
Acked-by: Jeff Mahoney <jeffm@suse.com>
---
 kernel/Kconfig.locks |   12 ++++++++++++
 1 file changed, 12 insertions(+)

--- a/kernel/Kconfig.locks
+++ b/kernel/Kconfig.locks
@@ -200,3 +200,15 @@ config INLINE_WRITE_UNLOCK_IRQRESTORE
 
 config MUTEX_SPIN_ON_OWNER
 	def_bool SMP && !DEBUG_MUTEXES
+
+#
+# The following option has kABI impact and should only be enabled on special
+# kernel flavor with limited distribution.
+#
+config MUTEX_SPIN_USE_MCS_QUEUE
+	bool "mutex: spin using mcs queue"
+	depends on MUTEX_SPIN_ON_OWNER
+	help
+	  This option enables MCS locking but is kABI-incompatible with
+	  the non-MCS locking variant.
+	bool
